### PR TITLE
small `api` changes for kustomize fix

### PR DESCRIPTION
**Pull request by natasha41575**
Some small changes that will be necessary to modify the `kustomize edit fix` commands to convert vars to replacements.

Essentially I'm just trying to reduce the size of #3926.

Related to https://github.com/kubernetes-sigs/kustomize/issues/3849

### CODE REVIEW

File: api/types/selector.go [hunk 1]
```diff
@@ -28,6 +28,18 @@ type Selector struct {
 	LabelSelector string `json:"labelSelector,omitempty" yaml:"labelSelector,omitempty"`
 }
 
+func (s *Selector) Copy() Selector {
```
Comment (monopole) [inline api/types/selector.go:31]: just
```
func (s *Selector) Copy() Selector { return *s }
```
Of course either version will panic if s is nil

File: api/filesys/fsondisk.go [hunk 2]
```diff
@@ -100,6 +100,19 @@ func (fsOnDisk) IsDir(name string) bool {
...
+// ReadDir delegates to os.ReadDir
+func (fsOnDisk) ReadDir(name string) ([]string, error) {
+	dirEntries, err := os.ReadDir(name)
+	if err != nil {
+		return nil, err
+	}
+	var result []string
+	for _, d := range dirEntries {
+		result = append(result, d.Name())
+	}
```
Comment (monopole) [inline api/filesys/fsondisk.go:112]: size of the result is known, so no need for the append
```
result := make(xx, len(dirEntries))
for i := range dirEntries {
  result[i] = dirEntries[i].Name
}
```

```

File: api/types/selector.go (hunk 1 above)
Comment (natasha41575) [reply api/types/selector.go:31]: done

File: api/filesys/fsondisk.go (hunk 2 above)
Comment (natasha41575) [reply api/filesys/fsondisk.go:112]: done, thank you

**Pull request merged**